### DISCUSSION TITLE
Attaching instructions to a lyric

### DISCUSSION THREAD

**Discussion by samuelbradshaw** [w3c-cg/musicxml#455] [General]
Is there any way in MusicXML to add instructions to a lyric element (or a line of lyrics)?

The song we need to encode has two choruses – the first chorus is used for verses 1–3, and the second chorus for verse 4. The format is compact (from a hymnal). We have a note above the second chorus indicating to the singer when the second chorus should be used. Here's a screenshot from a PDF exported from Finale:
<img width="385" alt="Screen Shot 2022-04-11 at 2 59 04 PM" src="https://user-images.githubusercontent.com/7323052/162831110-b54539dc-4290-4657-ae21-af61923186e7.png">

Right now, we're attaching the instructions (`<direction>` element in MusicXML) to the measure, and manually adjusting the vertical positioning of the instructions and lyrics so that the instructions appear in the right place.

Strict MusicXML renderers may be able to vertically position each text element the same as the source; however, the renderers we're using discard strict positioning in favor of defaults that adjust better for different screen sizes (which is a feature we rely on for our use cases). In the end, it looks something like this:

![IMG_A65C6BFBAA0C-1](https://user-images.githubusercontent.com/7323052/162836546-1eeff388-5423-4fd9-86fb-1b512527b20f.jpeg)

If there were a way to attach the instructions to a lyric, instead of generically attaching them to a measure, renderers that make their own layout decisions may be able to render instructions like this more cleanly. Any ideas?

For reference, here are the PDF and MusicXML version of the sheet music:
https://media.ldscdn.org/pdf/music/hymns/2001-01-2580-o-thou-rock-of-our-salvation-eng.pdf
[2001-01-2580-o-thou-rock-of-our-salvation-eng.musicxml.zip](https://github.com/w3c/musicxml/files/8467694/2001-01-2580-o-thou-rock-of-our-salvation-eng.musicxml.zip)

**Comment by adrianholovaty** [comment]
I'm interested in the answer here as well (and I'm reasonably sure there isn't a good answer yet, beyond "just treat it as plain text").

The second system in this screenshot is a somewhat related example, though this is spoken words as opposed to instructions:

![lyrics-example](https://user-images.githubusercontent.com/180401/163570470-f3e78373-1553-457d-b974-9584255bca4c.png)

Spoken lyrics like this are perhaps an even different thing (lyrics that are not attached to a specific note in the music), but the similarity is it's "lyric-adjacent" content that's intended to be rendered in the lyrics area of the notation.

**Comment by mdgood** [comment]
This has been brought up in issue #57 with a verse-number specific version at #7. Those are both still open issues.

This gets into an area of semantics vs appearance that may better be addressed by MNX rather than MusicXML. But if someone comes up with a design proposal that integrates well with MusicXML 4.0, we could certainly consider that for a future version.
